### PR TITLE
Fixes #705 - Throw error when the root HTML element has a dynamic id …

### LINKED ISSUE #705: passing data from state in the include tag
### Context
component state is out of scope inside the include tag

### Expected Behavior
pass state variable to included template

### Actual Behavior
state is undefined

### Steps to Reproduce
I have a component something like

```marko
class {
  onCreate() {
    this.state = { id: 1}
  }
}

//works fine
< input id=state.id  />

//state is undefined
<include(/path/to/generic/input, {id: state.id}) />
<include(/path/to/generic/input) id=state.id />
```

### Stack Trace
```
Uncaught ReferenceError: state is not defined
    at render (index.marko:40)
    at safeRender (renderable.js:6)
    at Template.render (renderable.js:136)
    at doInclude (include-tag.js:17)
    at includeTag (include-tag.js:30)
    at render (index.marko:45)
    at Template.renderSync (renderable.js:69)
    at Object.<anonymous> (index-desktop.marko.js:12)
    at __webpack_require__ (bootstrap e38ec24…:19)
    at Object.<anonymous> (emitter.js:2)
```

### PR DESCRIPTION
**Pull request by austinkelleher**
#705 /cc @patrick-steele-idem @mlrawlings

### CODE REVIEW

File: src/components/taglib/TransformHelper/handleRootNodes.js [hunk 1]
```diff
@@ -199,9 +199,16 @@ module.exports = function handleRootNodes() {
                     hasLegacyExplicitBind = true;
                 } else {
                     if (node.hasAttribute('id')) {
+                        let id = node.getAttributeValue('id');
+
+                        if (id && id.name) {
```
Comment (patrick-steele-idem) [inline src/components/taglib/TransformHelper/handleRootNodes.js]: The following check would be better:

```js
if (id && id.type !== 'Literal') {
```

Comment (patrick-steele-idem) [inline src/components/taglib/TransformHelper/handleRootNodes.js]: Also, please add a link to an error page with more explanation of the restriction. We are using the Wiki to host error pages: https://github.com/marko-js/marko/wiki/Error-Messages

Comment (austinkelleher) [inline src/components/taglib/TransformHelper/handleRootNodes.js]: Okay will do. Thanks.

**Comment by patrick-steele-idem** [comment]
@mlrawlings Can you please review and merge?

**Comment by austinkelleher** [comment]
Looks like the failing build is an unrelated timeout.

**Comment by austinkelleher** [comment]
Published `marko@4.4.8`

**Pull request merged**